src/settings/BatchGroupConfigurationSettings/BatchGroupConfigurationSettings.js

```diff
@@ -31,14 +84,40 @@ const BatchGroupConfigurationSettings = ({ mutator }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   []);
 
-  const onSaveExportConfig = useCallback(
-    () => {
+  useEffect(() => {
+    fetchExportConfig(selectedBatchGroupId);
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [selectedBatchGroupId]);
 
+  const onSave = useCallback(
+    (formValues) => {
+      saveExportConfig(formValues, mutator, selectedBatchGroupId, fetchExportConfig, showCallout);
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [selectedBatchGroupId],
   );
 
+  const selectBatchGroup = useCallback(
+    (id) => {
+      setSelectedBatchGroupId(id);
+    }, [],
+  );
+
+  const initialValues = exportConfig?.id
```

Comment:
I'm not sure in this approach, since every render you'll get a new object. From my perspective initialValues should be built once, when you have all the required data, saved in state for example and only after this should happen render of the form.

Reply:
it depends, if form object is updated I don't see reason to not change initial values

I think it's a case

Reply:
but it should be the same if nothing is changed

Reply:
so you mean it Form's responsibility to handle changes or even new object in initialValues prop every render? It makes sense

Reply:
yep, and as I mentioned before it should be the same object if nothing is changed, here each render will be new object